romio: actually set status in MPI_File_read_all

hzhou opened the pull request:
## Pull Request Description

The old code blindly sets status of MPI_File_read_all with the input
data size. This ignore the cases when the actual file is too short or
have issues of actually getting the requested data. This patch updates
the status with actual number of bytes received.

Fixes #1085



## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

hzhou commented:
test:mpich/custom
testlist: io

roblatham00 commented:
I pushed some warning fixes to your branch

roblatham00 reviewed (approved):
Thanks for digging into some hairy code.   Approved

hzhou commented:
test:mpich/ch3/tcp
test:mpich/ch4/most

hzhou commented:
test:mpich/ch3/tcp
test:mpich/ch4/most

hzhou commented:
test:mpich/ch3/tcp
test:mpich/ch4/most

hzhou commented:
Added warning-fix for -Wmaybe-uninitialized (from https://github.com/pmodels/mpich/pull/6114)
Fixed memory leak.
Retest-
test:mpich/ch3/tcp
test:mpich/ch4/most
